File: number_of_steps_to_reduce_a_number_to_zero/number_of_steps.py
def numberOfSteps(num):
    if num >= 0:
        steps = 0
        while num != 0:
            if (num % 2) == 0:
                num /= 2
                steps += 1
            else:
                num -= 1
                steps += 1
        return steps
    else:
        return 'Number must be greater than zero'

print(numberOfSteps(14))
print(numberOfSteps(123))
print(numberOfSteps(0))
print(numberOfSteps(-5))

File: number_of_steps_to_reduce_a_number_to_zero/number_of_steps.rb
def number_of_steps(num)
    if num >= 0
        steps = 0
        while num != 0
            if num.even?
                num /= 2
                steps += 1
            else
                num -= 1
                steps += 1
            end
        end
        return steps
    else
        return 'Number must be greater than zero'
    end
end

puts number_of_steps(14)
puts number_of_steps(123)
puts number_of_steps(0)
puts number_of_steps(-5)